python/packages/ai/teams/ai/models/openai_model.py

```diff
@@ -0,0 +1,205 @@
+        return self._options
+
+    def __init__(self, options: Union[OpenAIModelOptions, AzureOpenAIModelOptions]) -> None:
+        """
+        Creates a new `OpenAIModel` instance.
+
+        Args:
+            options (OpenAIModelOptions | AzureOpenAIModelOptions): model options.
+        """
+
+        self._options = options
+
+        if isinstance(options, OpenAIModelOptions):
+            self._client = openai.AsyncOpenAI(
+                api_key=options.api_key,
+                base_url=options.endpoint,
+                organization=options.organization,
+                default_headers={"User-Agent": "teamsai-py/1.0.0"},
+            )
+        elif isinstance(options, AzureOpenAIModelOptions):
+            self._client = openai.AsyncAzureOpenAI(
+                api_key=options.api_key,
+                api_version=options.api_version,
+                azure_endpoint=options.endpoint,
+                azure_deployment=options.default_model,
+                organization=options.organization,
+                default_headers={"User-Agent": "teamsai-py/1.0.0"},
+            )
+
+    async def complete_prompt(
+        self,
+        context: TurnContext,
+        memory: Memory,
+        functions: PromptFunctions,
+        tokenizer: Tokenizer,
+        template: PromptTemplate,
+    ) -> PromptResponse[str]:
+        max_tokens = template.config.completion.max_input_tokens
+        model = (
+            template.config.completion.model
+            if template.config.completion.model is not None
+            else self._options.default_model
+        )
+        res = await template.prompt.render_as_messages(
+            context=context,
+            memory=memory,
+            functions=functions,
+            tokenizer=tokenizer,
+            max_tokens=max_tokens,
+        )
+
+        if res.too_long:
+            return PromptResponse[str](
+                status="too_long",
+                error=f"""
+                the generated chat completion prompt had a length of {res.length} tokens 
+                which exceeded the max_input_tokens of {max_tokens}
+                """,
+            )
+
+        if self._options.logger is not None:
+            self._options.logger.debug(f"PROMPT:\n{res.output}")
+
+        messages: List[chat.ChatCompletionMessageParam] = []
+
+        for msg in res.output:
+            param: chat.ChatCompletionMessageParam
+
+            if msg.role == "user":
+                param = chat.ChatCompletionUserMessageParam(
+                    role="user",
+                    name=msg.name if msg.name is not None else "",
+                    content=msg.content if msg.content is not None else "",
+                )
+            elif msg.role == "assistant":
+                param = chat.ChatCompletionAssistantMessageParam(
+                    role="assistant",
+                    name=msg.name if msg.name is not None else "",
+                    content=msg.content,
```

Comment:
Can msg.content be of type ' None' here?

Reply:
yes, but this openai data type allows content type None

Reply:
https://github.com/openai/openai-python/blob/main/src/openai/types/chat/chat_completion_assistant_message_param.py#L30